lib/failure.sh

```diff
@@ -445,6 +445,25 @@ fail_missing_yarn_vendor() {
   fi
 }
 
+fail_corepack_not_available() {
+  local package_manager="$1"
+  local node_version="$2"
+
+  mcount "failures.corepack-unsupported"
+  meta_set "failure" "failures.corepack-unsupported"
+  header "Build failed"
+  warn "Corepack is not supported in Node.js $node_version
+
+    Your application indicated that $package_manager should be installed using Corepack. This feature
+    is included with all Node.js releases starting from Node.js 14.19.0 / 16.9.0. The version
+    of Node.js used in this build is $node_version which does not support Corepack.
+
+    To use Corepack, update your Node.js version:
```

Comment:
This action is one way to fix the issue. I think there is another. Something like:

> To use $package_manager without Corepack, remove `packageManager` from package.json. Specify the desired $package_manager version in `package.json` `engines` instead.

I don't know if we have to document this, though. We only support Node.js versions that have Corepack bundled. So anyone in this scenario is already on an unsupported Node.js.

Reply:
I consider this an extreme edge case because if you're using a version of Node.js that is this out-of-date, it's unlikely you'd be configured for corepack use.  So, I figured if a user gets this error, it's more likely they are trying to opt-in to corepack so it's a good opportunity to prompt them to update their Node.js version 😅.

Do you think there's value in adding the secondary action of removing the `packageManager` field?

Reply:
I agree; I'm happy without the additional message.